agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -1,164 +1,90 @@
 package randoop.instrument;
 
-import static java.lang.System.out;
-
 import java.io.ByteArrayInputStream;
-import java.io.File;
-import java.io.FileReader;
-import java.io.IOException;
-import java.io.LineNumberReader;
 import java.lang.instrument.ClassFileTransformer;
 import java.lang.instrument.IllegalClassFormatException;
 import java.security.ProtectionDomain;
-import java.util.ArrayList;
-import java.util.LinkedHashMap;
-import java.util.List;
 import java.util.Map;
-import java.util.regex.Pattern;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 import org.apache.bcel.Const;
-import org.apache.bcel.classfile.Attribute;
 import org.apache.bcel.classfile.ClassParser;
-import org.apache.bcel.classfile.Constant;
-import org.apache.bcel.classfile.ConstantUtf8;
 import org.apache.bcel.classfile.JavaClass;
 import org.apache.bcel.classfile.Method;
 import org.apache.bcel.generic.ClassGen;
 import org.apache.bcel.generic.CodeExceptionGen;
 import org.apache.bcel.generic.ConstantPoolGen;
-import org.apache.bcel.generic.INVOKESTATIC;
-import org.apache.bcel.generic.INVOKEVIRTUAL;
 import org.apache.bcel.generic.Instruction;
 import org.apache.bcel.generic.InstructionFactory;
 import org.apache.bcel.generic.InstructionHandle;
 import org.apache.bcel.generic.InstructionList;
 import org.apache.bcel.generic.InstructionTargeter;
+import org.apache.bcel.generic.InvokeInstruction;
 import org.apache.bcel.generic.LineNumberGen;
 import org.apache.bcel.generic.LocalVariableGen;
 import org.apache.bcel.generic.MethodGen;
 import org.apache.bcel.generic.Type;
-import plume.ArraysMDE;
 import plume.BCELUtil;
 import plume.SimpleLog;
-import plume.StrTok;
-import plume.UtilMDE;
 
 /**
- * The Instrument class is responsible for modifying another class's bytecode. It changes calls to
- * specified functions into static calls to user functions. This allows a user of Randoop to capture
- * arbitrary calls and modify them as desired. For example, all calls to
- * JOptionPane.showConfirmDialog (which would otherwise hang a regressions test) can be changed to
- * throw an exception, or simply print a message and continue.
+ * The {@code CallReplacementTransformer} replaces each call to method m1 by a call to method m2. It
+ * is used by the MapCallsAgent.
+ *
+ * <p>Replacements may be given as original-replacement pairs of methods, classes, or packages. For
```

Comment:
This text seems irrelevant, because this class operates on a map.  The paragraph is also incomplete, with a final sentence that trails off.